### ISSUE TITLE
Authenticator SignIn with Vue.js - user attributes missing

### ISSUE THREAD

**Issue by jonathanlb** [aws-amplify/amplify-ui#1532]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

Vue

### Which UI component?

Authenticator

### How is your app built?

@vue/cli-service 5.0.3

### Please describe your bug.

Similar to issue aws-amplify/amplify-ui#1270 but with Vue.js (v 3.2.31), when rendering the Authenticator widget immediately after the user logs in for the first time, verifies attributes, and overwrites the initial temporary password, the fields under the `user` property are undefined.

I'm using npm versions of aws-amplify/ui-vue 2.3.1 and aws-amplify/auth 4.4.3.

### What's the expected behaviour?

The expected behavior in the example below is to see the text "Hello <user name>", but the `attributes.name` field is undefined during the first login session.

If the user logs out or reloads the page, the attributes are defined.

### Help us reproduce the bug!

- Create a new user in a Cognito user pool.
- Present an Authenticator Vue.js component wired to the Cognito user pool.  Render `user.attributes.name`.
- Enter the new username/email and temporary password.

### Code Snippet

```vue
<template>
  <authenticator :hide-sign-up="true">
    <template v-slot="{ user, signOut }">
      <p>Hello {{ user?.attributes?.name }}!</p>
      <button @click="signOut">Sign Out</button>
    </template>
  </authenticator>
</template>

<script>
import { Authenticator } from '@aws-amplify/ui-vue';
import '@aws-amplify/ui-vue/styles.css';

import Amplify from 'aws-amplify';
import awsconfig from './aws-exports';

Amplify.configure(awsconfig);

export default {
  name: 'App',
  components: {
    Authenticator
  }
}
</script>
```
```


### Additional information and screenshots

package.json

```
{
  "name": "amplify-demo",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "serve": "vue-cli-service serve",
    "build": "vue-cli-service build",
    "lint": "vue-cli-service lint",
    "test": "jest"
  },
  "dependencies": {
    "@aws-amplify/auth": "^4.4.3",
    "@aws-amplify/storage": "^4.4.17",
    "@aws-amplify/ui-vue": "^2.3.1",
    "core-js": "^3.8.3",
    "debug": "^4.3.3",
    "vue": "^3.2.31"
  },
  "devDependencies": {
    "@babel/core": "^7.12.16",
    "@babel/eslint-parser": "^7.12.16",
    "@vue/cli-plugin-babel": "~5.0.0",
    "@vue/cli-plugin-eslint": "~5.0.0",
    "@vue/cli-service": "~5.0.3",
    "@vue/test-utils": "^2.0.0-rc.17",
    "@vue/vue3-jest": "^27.0.0-alpha.4",
    "aws-amplify": "^4.3.16",
    "eslint": "^7.32.0",
    "eslint-plugin-vue": "^8.0.3",
    "jest": "^27.5.1"
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/vue3-essential",
      "eslint:recommended"
    ],
    "parserOptions": {
      "parser": "@babel/eslint-parser"
    },
    "rules": {}
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not dead",
    "not ie 11"
  ]
}
```

**Comment by jonathanlb** [comment]
Yes, the problem only occurs during the first login session after the user enters the name and new password, but before the user logs out or reloads the page.

**Comment by wlee221** [comment]
Thanks for detailed repro steps, I could repro this! Taking a look.

**Comment by wlee221** [comment]
@jonathanlb, we just merged the fix #1643 to `@aws-amplify/ui-vue@2.3.8`. Can you try upgrading to latest and let us know if the problem resolved?

**Comment by jonathanlb** [comment]
Yes, thank you, @wlee221 .  The problem was resolved upon updating to `@aws-amplify/ui-vue@2.3.8`.
